lib/logstash/inputs/kinesis/worker.rb

```diff
@@ -12,18 +14,36 @@ class LogStash::Inputs::Kinesis::Worker
   def initialize(*args)
     # nasty hack, because this is the name of a method on IRecordProcessor, but also ruby's constructor
     if !@constructed
-      @codec, @output_queue, @decorator, @checkpoint_interval, @logger = args
+      @codec, @output_queue, @decorator, @checkpoint_interval, @compression,
+        @cloudwatch_logs, @logger = args
+
+      raise(
+        LogStash::ConfigurationError,
+        'cloudwatch_logs requires "gzip" compression'
+      ) if cloudwatch_logs && compression != 'gzip'
+
+      @processor = case compression
+                   when nil
+                     require 'logstash/inputs/kinesis/processors/charset'
+                     LogStash::Inputs::Kinesis::Processors::Charset.new
+                   when 'gzip'
+                     require 'logstash/inputs/kinesis/processors/gzip'
+                     LogStash::Inputs::Kinesis::Processors::GZip.new
+                   end
```

Comment:
let's add an `else` clause and raise a `ConfigurationError` if any other (invalid) value is specified for `compression`